__tests__/integration.js

```diff
@@ -232,7 +232,9 @@ describe('--registry option', () => {
       await runYarn(['add', 'is-array', '--registry', registry], {cwd});
     } catch (err) {
       const stdoutOutput = err.message;
-      expect(stdoutOutput.toString()).toMatch(/getaddrinfo ENOTFOUND example-registry-doesnt-exist\.com/g);
+      expect(stdoutOutput.toString()).toMatch(
+        /getaddrinfo ENOTFOUND example-registry-doesnt-exist\.com|example-registry-doesnt-exist.com\/is-array: connect ECONNREFUSED/,
```

Comment:
This was necessary in order for tests to pass locally, for me. Yet, it is still valid because it can depend on several network related events. In my case, it's probably my ISP.

Reply:
to elaborate, my ISP's DNS servers will return a canned IP address for any invalid lookups that directs you to some static search page that they have set up, so you get an ECONNREFUSED instead of the ENOTFOUND.